### PR TITLE
Update btcd dependency, add Signet params

### PR DESCRIPTION
**Pull request by guggero**
Required for https://github.com/lightningnetwork/lnd/pull/5025.

Depends on https://github.com/btcsuite/btcd/pull/1692.

Adds support for Bitcoin Signet.

cc @Roasbeef, @wpaulino.

### CODE REVIEW

**Review by wpaulino — COMMENTED** [review]
LGTM, travis is failing with:

```
--- FAIL: TestRPCMethodHelpGeneration (0.01s)
    rpchelp_test.go:59: Cannot generate 'en_US' help for method 'listsinceblock': missing description for 'listtransactionsresult-label'
    rpchelp_test.go:59: Cannot generate 'en_US' help for method 'listtransactions': missing description for 'listtransactionsresult-label'
    rpchelp_test.go:59: Cannot generate 'en_US' help for method 'listaddresstransactions': missing description for 'listtransactionsresult-label'
    rpchelp_test.go:59: Cannot generate 'en_US' help for method 'listalltransactions': missing description for 'listtransactionsresult-label'
    rpchelp_test.go:46: Regenerate help texts with 'go generate' after fixing
```

**Comment by guggero** [comment]
Fixed Travis.

**Comment by guggero** [comment]
Rebased. @Roasbeef PTAL.

File: go.mod [hunk 1]
```diff
@@ -36,4 +36,14 @@ replace github.com/btcsuite/btcwallet/wallet/txrules => ./wallet/txrules
 
 replace github.com/btcsuite/btcwallet/wallet/txsizes => ./wallet/txsizes
 
+replace (
+	github.com/btcsuite/btcd => github.com/guggero/btcd v0.20.1-beta.0.20210422111045-7d1ab0b4d769
```
Comment (Roasbeef) [inline go.mod:40]: Can point to mainline `btcd` now after the PR merge.

**Review by Roasbeef — APPROVED** [review]
LGTM 🥃

File: internal/legacy/keystore/keystore.go [hunk 2]
```diff
@@ -476,6 +476,8 @@ func (net *netParams) ReadFrom(r io.Reader) (int64, error) {
 		*net = (netParams)(chaincfg.TestNet3Params)
 	case wire.SimNet:
 		*net = (netParams)(chaincfg.SimNetParams)
+	case chaincfg.SigNetParams.Net:
```
Comment (wpaulino) [inline internal/legacy/keystore/keystore.go:479]: We'll also want to handle this in a few more places in `waddrmgr` after the account import PRs.

**Comment by guggero** [comment]
Rebased and added a few more cases that were previously not covered.

**Review by Roasbeef — APPROVED** [review]
LGTM 🛎

**Pull request merged**